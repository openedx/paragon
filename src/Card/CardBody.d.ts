import React from 'react';

export interface CardBodyProps {
  className?: string,
  children?: React.ReactNode,
}
declare const CardBody: React.ForwardRefExoticComponent<CardBodyProps>;
export default CardBody;
