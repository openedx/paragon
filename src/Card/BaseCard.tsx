import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import type { ComponentWithAsProp, BsPropsWithAs } from '../utils/types/bootstrap';

// @ts-ignore
import CardBody from './CardBody';

const BASE_CARD_CLASSNAME = 'card';

const colorVariants = [
  'primary',
  'secondary',
  'success',
  'danger',
  'warning',
  'info',
  'dark',
  'light',
] as const;

const textVariants = [
  'white',
  'muted',
] as const;

type ColorVariant = typeof colorVariants[number];
type TextVariant = typeof textVariants[number];
interface Props extends BsPropsWithAs {
  prefix?: string;
  bgColor?: ColorVariant;
  textColor?: ColorVariant | TextVariant;
  borderColor?: ColorVariant;
  hasBody?: boolean;
  className?: string;
  children: React.ReactNode;
}
type BaseCardType = ComponentWithAsProp<'div', Props>;

const BaseCard : BaseCardType = React.forwardRef<HTMLDivElement, Props>(
  (
    {
      prefix,
      className,
      bgColor,
      textColor,
      borderColor,
      hasBody = false,
      children,
      as: Component = 'div',
      ...props
    },
    ref,
  ) => {
    const classes = classNames(
      className,
      prefix ? `${prefix}-${BASE_CARD_CLASSNAME}` : BASE_CARD_CLASSNAME,
      bgColor && `bg-${bgColor}`,
      textColor && `text-${textColor}`,
      borderColor && `border-${borderColor}`,
    );

    return (
      <Component ref={ref} {...props} className={classes}>
        {hasBody ? <CardBody>{children}</CardBody> : children}
      </Component>
    );
  },
);

/* eslint-disable react/require-default-props */
BaseCard.propTypes = {
  /** Prefix for component CSS classes. */
  prefix: PropTypes.string,
  /** Background color of the card. */
  bgColor: PropTypes.oneOf(colorVariants),
  /** Text color of the card. */
  textColor: PropTypes.oneOf([...colorVariants, ...textVariants]),
  /** Border color of the card. */
  borderColor: PropTypes.oneOf(colorVariants),
  /** Determines whether the card should render its children inside a `CardBody` wrapper. */
  hasBody: PropTypes.bool,
  /** Set a custom element for this component. */
  as: PropTypes.elementType,
  /** Additional CSS class names to apply to the card element. */
  className: PropTypes.string,
  /** The content to render inside the card. */
  children: PropTypes.node,
};

export default BaseCard;
