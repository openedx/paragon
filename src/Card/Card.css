/**
 * Do not edit directly
 * Generated on Wed, 01 Mar 2023 00:27:19 GMT
 */

:root {
  --pgn-spacing-card-focus-border-offset: 5px;
  --pgn-spacing-card-logo-bottom-offset-horizontal: .4375rem;
  --pgn-spacing-card-logo-bottom-offset-base: 1rem;
  --pgn-spacing-card-logo-left-offset-horizontal: .4375rem;
  --pgn-spacing-card-logo-left-offset-base: 1.5rem;
  --pgn-spacing-card-loading-skeleton-spacer: .313rem;
  --pgn-spacing-card-footer-action-gap: .5rem;
  --pgn-spacing-card-columns-gap: 1.25rem;
  --pgn-spacing-card-columns-count: 3;
  --pgn-spacing-card-margin-group: 12px;
  --pgn-spacing-card-spacer-y: .75rem;
  --pgn-spacing-card-spacer-x: 1.25rem;
  --pgn-size-card-logo-height: 4.125rem;
  --pgn-size-card-logo-width: 7.25rem;
  --pgn-size-card-image-vertical-max-height: 140px;
  --pgn-size-card-image-horizontal-width-max: 240px;
  --pgn-size-card-focus-border-radius: 5px;
  --pgn-size-card-focus-border-width: 2px;
  --pgn-size-card-border-radius-logo: .25rem;
  --pgn-size-card-border-radius-image: .3125rem;
  --pgn-typography-footer-text-font-size: var(--pgn-typography-font-size-small-x);
  --pgn-spacing-card-columns-margin: var(--pgn-spacing-card-spacer-y);
  --pgn-spacing-card-margin-grid: var(--pgn-spacing-card-margin-group);
  --pgn-spacing-card-margin-deck: var(--pgn-spacing-card-margin-group);
  --pgn-size-card-image-horizontal-width-min: var(--pgn-size-card-image-horizontal-width-max);
  --pgn-size-card-border-radius-base: var(--pgn-size-border-radius-base);
  --pgn-size-card-border-width: var(--pgn-size-border-width);
  --pgn-spacing-card-margin-grid-bottom: var(--pgn-spacing-spacer-3);
  --pgn-spacing-card-margin-deck-bottom: var(--pgn-spacing-spacer-3);
}
