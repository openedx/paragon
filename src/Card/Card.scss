@import "variables";
@import "~bootstrap/scss/card";

.pgn__card,
.pgn__card-body {
  width: 100%;
  border: 0;

  %dark-variant-content {
    color: $white;

    a {
      color: $white;
    }
  }

  &.pgn__card-dark {
    background-color: $card-bg-dark;

    @extend %dark-variant-content;

    .pgn__card-header {
      @extend %dark-variant-content;

      .pgn__card-header-title,
      .pgn__card-header-title-sm,
      .pgn__card-header-title-md,
      .pgn__card-header-subtitle,
      .pgn__card-header-subtitle-sm,
      .pgn__card-header-subtitle-md {
        @extend %dark-variant-content;
      }
    }

    .pgn__card-section {
      @extend %dark-variant-content;

      .pgn__card-section-title {
        @extend %dark-variant-content;
      }
    }

    .pgn__card-footer,
    .pgn__card-footer-text {
      @extend %dark-variant-content;
    }
  }

  &.pgn__card-muted {
    background-color: $card-bg-muted;
  }
}

a .pgn__card {
  color: $gray-700;
  display: inline-block;
}

.pgn__card-grid {
  .row > div[class*="col-"] {
    margin-bottom: $card-grid-margin-bottom;
  }
}

.pgn__card-grid,
.pgn__card-deck {
  .row > div[class*="col-"] {
    display: flex;
    flex: 1 0 auto;

    &.pgn__card__disable-equal-column-heights {
      display: block;
    }
  }
}

.pgn__card-deck {
  position: relative;
  margin-bottom: $card-deck-margin-bottom;

  &.pgn__card-deck-has-horizontal-scroll {
    .pgn__card-deck-row {
      overflow-x: scroll;
    }
  }

  .pgn__card-deck-row {
    overflow: hidden;
    flex-wrap: nowrap;
    padding-top: $card-focus-border-offset;
    padding-bottom: $card-focus-border-offset;

    &:focus-visible {
      outline: $card-focus-border-width solid $card-border-focus-color;
      border-radius: $card-focus-border-radius;
    }
  }
}

.pgn__card-carousel {
  .pgn__card-carousel-title {
    margin-bottom: map-get($spacers, 2);
  }

  .pgn__card-carousel-subtitle {
    font-size: $font-size-sm;
    margin-bottom: map-get($spacers, 2);
  }

  .pgn__card-carousel-header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: map-get($spacers, 1);
  }
}

.pgn__card-header {
  padding: 0 $card-spacer-x;
  display: flex;
  justify-content: space-between;

  .pgn__card-header-content {
    display: flex;
    flex-direction: column;
    margin-top: map_get($spacers, 4\.5);
    overflow: auto;
    text-align: start;
    width: 100%;
  }

  %header-title {
    color: $black;
    font-weight: $font-weight-bold;
    text-align: start;
  }

  .pgn__card-header-title,
  .pgn__card-header-title-sm,
  .pgn__card-header-title-md {
    @extend %header-title;
  }

  .pgn__card-header-title-sm {
    font-size: $h4-font-size;
  }

  .pgn__card-header-title-md {
    font-size: $h3-font-size;
  }

  %header-subtitle {
    color: $gray-700;
    margin-top: map_get($spacers, 1);
  }

  .pgn__card-header-subtitle,
  .pgn__card-header-subtitle-sm,
  .pgn__card-header-subtitle-md {
    @extend %header-subtitle;
  }

  .pgn__card-header-subtitle-sm {
    font-size: $h5-font-size;
  }

  .pgn__card-header-subtitle-md {
    font-size: $h4-font-size;
  }

  .pgn__card-header-actions {
    margin-top: $spacer;
    margin-inline-start: $spacer;
    flex-shrink: 0;
  }

  .pgn__card-header-loader {
    width: 100%;

    .react-loading-skeleton {
      margin-top: map-get($spacers, 4);
    }
  }
}

.pgn__card-divider {
  border-top: 1px solid $card-divider-bg;
  height: 0;
}

.pgn__card-section {
  padding: $card-spacer-x;
  flex-grow: 1;
  text-align: start;

  &.is-muted {
    background-color: $light-200;
  }

  .pgn__card-section-title {
    color: $black;
    font-weight: $font-weight-bold;
    font-size: $h5-font-size;
    margin-bottom: $card-spacer-y;
  }

  .pgn__card-section-actions {
    display: flex;
    justify-content: flex-end;
    margin-top: $card-spacer-y;
  }

  dd {
    [dir="rtl"] & {
      margin-right: 0;
      margin-left: 40px;
    }
  }
}

.pgn__card-footer {
  width: 100%;
  padding: $card-spacer-x;
  display: flex;
  align-items: center;
  justify-content: center;

  &.vertical {
    justify-content: flex-end;
    padding-top: 0;

    & > * + * {
      margin-inline-start: $card-footer-actions-gap;
    }
  }

  .pgn__card-footer-loader {
    width: 100%;
    margin-inline-start: 0;
  }

  &.vertical-stacked {
    flex-direction: column-reverse;
    padding-top: $card-spacer-x - $card-footer-actions-gap;

    & > * {
      width: 100%;
      margin-top: $card-footer-actions-gap;
    }
  }

  &.horizontal {
    flex-wrap: wrap;

    & > :not(:last-child) {
      margin-inline-end: $card-footer-actions-gap;
    }
  }

  &.horizontal-stacked {
    flex-direction: column-reverse;
    padding-top: $card-spacer-x - $card-footer-actions-gap;

    & > * {
      margin-top: $card-footer-actions-gap;
    }
  }
}

.pgn__card-footer-text {
  color: $gray-700;
  order: 1;
  font-size: $card-footer-text-font-size;

  &.vertical {
    margin-inline-end: auto;
    order: -1;
  }

  &.vertical-stacked {
    margin-right: auto;
  }

  &.horizontal {
    margin-top: $card-spacer-x;
    width: 100%;
    text-align: center;
  }

  &.horizontal-stacked {
    margin-top: $card-spacer-x;
    width: 100%;
    text-align: center;
    order: -1;
  }
}

.pgn__card-wrapper-image-cap {
  position: relative;

  &.horizontal {
    max-width: $card-image-horizontal-max-width;
    min-width: $card-image-horizontal-min-width;
    overflow: hidden;

    .pgn__card-image-cap {
      height: 100%;
      max-width: inherit;
      width: auto;
      object-fit: cover;
    }

    .pgn__card-logo-cap {
      inset-inline-start: $card-logo-left-offset-horizontal;
      bottom: $card-logo-bottom-offset-horizontal;
    }

    .pgn__card-image-cap-loader {
      .react-loading-skeleton {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
      }
    }
  }

  &.vertical {
    max-height: $card-image-vertical-max-height;
  }

  .pgn__card-image-cap-loader {
    .react-loading-skeleton {
      margin-bottom: -$loading-skeleton-spacer;
      position: relative;
      top: -$loading-skeleton-spacer;
      height: 100%;
    }
  }
}

.pgn__card-image-cap {
  object-fit: cover;
  max-height: inherit;
  width: 100%;
  display: none;

  &.show {
    display: block;
  }
}

.pgn__card-logo-cap {
  position: absolute;
  inset-inline-start: $card-logo-left-offset;
  bottom: #{-$card-logo-bottom-offset};
  width: $card-logo-width;
  height: $card-logo-height;
  border-radius: $card-logo-border-radius;
  box-shadow: $level-1-box-shadow;
  padding: map_get($spacers, 2);
  background-color: $white;
  display: none;

  &.show {
    display: block;
  }
}

%pgn__card-focused {
  outline: none;

  &::before {
    content: "";
    position: absolute;
    top: calc(-1 * #{$card-focus-border-offset});
    right: calc(-1 * #{$card-focus-border-offset});
    bottom: calc(-1 * #{$card-focus-border-offset});
    left: calc(-1 * #{$card-focus-border-offset});
    border: solid $card-focus-border-width $card-border-focus-color;
    border-radius: $card-focus-border-radius;
  }

  &.pgn__card-dark::before {
    border: solid $card-focus-border-width $card-border-focus-color-dark;
  }
}

.pgn__card {
  outline: none;
  overflow: hidden;

  @include pgn-box-shadow(1, "down");

  &.clickable {
    &:hover,
    &:focus,
    &.focus {
      cursor: pointer;

      @include pgn-box-shadow(2, "down");
    }

    &:focus,
    &.focus {
      @extend %pgn__card-focused;
    }
  }

  &.horizontal {
    flex-direction: row;
  }
}

.pgn__card-status {
  box-shadow: none;
  margin-bottom: 0;
  font-size: $font-size-sm;
  line-height: 1.5rem;
  text-align: start;
  position: relative;
  padding: $card-spacer-y $card-spacer-x;
  border: 0 solid transparent;
  border-radius: 0 0 $card-border-radius $card-border-radius;

  .pgn__card-status__content {
    display: flex;
  }

  .pgn__card-status__message-content {
    flex-grow: 1;

    & > :last-child {
      margin-bottom: 0;
    }
  }

  .pgn__card-status__actions {
    margin-top: map-get($spacers, 3);
  }

  .pgn__card-status__heading {
    font-size: $h4-font-size;
    color: $black;
    display: flex;
    font-weight: $font-weight-bold;
    line-height: 1.5rem;
    margin-bottom: .5rem;
  }

  .pgn__card-status__content-icon {
    margin-inline-end: .8rem;
    width: 1.5rem;
    height: 1.5rem;
    flex-shrink: 0;
  }

  @each $color, $value in $theme-colors {
    &.pgn__card-status__#{$color} {
      color: inherit;
      background-color: theme-color($color, "background");
      border-color: theme-color($color, "border");

      .pgn__card-status__content-icon {
        color: theme-color($color, "default");
      }
    }
  }

  &.pgn__card-status__primary {
    color: $white;
    background-color: $primary-500;
    border-color: $primary-500;
    box-shadow: 0 0 0 1px $primary-500;

    .pgn__card-status__content-icon {
      color: $white;
    }
  }
}

@media (max-width: map-get($grid-breakpoints, "sm")) {
  .pgn__card-footer {
    &.horizontal > :not(:last-child) {
      margin-inline-end: 0;
    }

    .btn {
      width: 100%;
      margin: .625rem 0 0;
      margin-inline-end: 0;
    }
  }

  .pgn__action-row {
    .btn:not(:last-child) {
      margin-bottom: .3125rem;
    }

    & > * + * {
      margin-inline-start: 0;
    }
  }

  .pgn__action-row-stacked,
  .pgn__card-section {
    .btn {
      width: 100%;
    }
  }
}
