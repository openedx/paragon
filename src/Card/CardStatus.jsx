import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Skeleton from 'react-loading-skeleton';
import Icon from '../Icon';
import CardContext from './CardContext';

const CardStatus = React.forwardRef(({
  className,
  children,
  variant,
  icon,
  title,
  actions,
}, ref) => {
  const { isLoading } = useContext(CardContext);

  if (isLoading) {
    return (
      <div
        className={classNames(
          'pgn__card-status',
          className,
        )}
        data-testid="card-status-skeleton"
        ref={ref}
      >
        <Skeleton />
      </div>
    );
  }

  return (
    <div
      className={classNames(
        'pgn__card-status',
        `pgn__card-status__${variant}`,
        className,
      )}
      ref={ref}
    >
      <div className="pgn__card-status__content">
        {icon && <Icon className="pgn__card-status__content-icon" src={icon} />}
        <div className="pgn__card-status__message-content">
          {title && <div className="pgn__card-status__heading">{title}</div>}
          {children}
        </div>
      </div>
      {!!actions && (
        <div className="pgn__card-status__actions">
          {actions}
        </div>
      )}
    </div>
  );
});

CardStatus.propTypes = {
  /** Specifies the content of the component. */
  children: PropTypes.node.isRequired,
  /** The class to append to the base element. */
  className: PropTypes.string,
  /** Icon that will be shown in the top-left corner. */
  icon: PropTypes.func,
  /** Specifies variant to use. */
  variant: PropTypes.oneOf([
    'primary',
    'success',
    'danger',
    'warning',
  ]),
  /** Specifies title for the `Card.Status`. */
  title: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  /** Specifies any optional actions, e.g. button(s). */
  actions: PropTypes.node,
};

CardStatus.defaultProps = {
  className: undefined,
  icon: undefined,
  variant: 'warning',
  title: undefined,
  actions: undefined,
};

export default CardStatus;
