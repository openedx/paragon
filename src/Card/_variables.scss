// Card

$card-spacer-y:                      .75rem !default;
$card-spacer-x:                      1.25rem !default;
$card-border-width:                  $border-width !default;
$card-border-radius:                 $border-radius !default;
$card-border-color:                  rgba($black, .125) !default;
$card-border-focus-color:            rgba($black, .5) !default;
$card-border-focus-color-dark:       theme-color("primary", "focus") !default;
$card-inner-border-radius:           subtract($card-border-radius, $card-border-width) !default;
$card-cap-bg:                        rgba($black, .03) !default;
$card-cap-color:                     null !default;
$card-height:                        null !default;
$card-color:                         null !default;
$card-bg:                            $white !default;
$card-bg-dark:                       $primary-500 !default;
$card-bg-muted:                      $light-200 !default;

$card-img-overlay-padding:           1.25rem !default;

$card-group-margin:                  calc($grid-gutter-width / 2) !default;
$card-deck-margin:                   $card-group-margin !default;
$card-grid-margin:                   $card-group-margin !default;
$card-deck-margin-bottom:            map_get($spacers, 3) !default;
$card-grid-margin-bottom:            map_get($spacers, 3) !default;

$card-columns-count:                 3 !default;
$card-columns-gap:                   1.25rem !default;
$card-columns-margin:                $card-spacer-y !default;

$card-divider-bg:                    $light-400 !default;
$card-divider-margin-y:              $card-spacer-y !default;

$card-footer-actions-gap:            .5rem !default;

$card-logo-left-offset:              1.5rem !default;
$card-logo-bottom-offset:            1rem !default;
$card-logo-left-offset-horizontal:   .4375rem !default;
$card-logo-bottom-offset-horizontal: .4375rem !default;

$card-logo-width:                    7.25rem !default;
$card-logo-height:                   4.125rem !default;

$card-image-border-radius:           .3125rem !default;
$card-logo-border-radius:            .25rem !default;

$card-footer-text-font-size:         $x-small-font-size;

$card-image-horizontal-max-width:    240px !default;
$card-image-horizontal-min-width:    $card-image-horizontal-max-width !default;
$card-image-vertical-max-height:     140px !default;
$loading-skeleton-spacer:            .313rem !default;

$card-focus-border-offset:           5px !default;
$card-focus-border-width:            2px !default;
$card-focus-border-radius:           5px !default;
