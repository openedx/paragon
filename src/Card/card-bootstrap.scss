.card {
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: $card-bg;
  background-clip: border-box;
  border: $card-border-width solid $card-border-color;
  position: relative;

  @include border-radius($card-border-radius);

  > hr {
    margin-right: 0;
    margin-left: 0;
  }

  > .list-group {
    border-top: inherit;
    border-bottom: inherit;

    &:first-child {
      border-top-width: 0;

      @include border-top-radius($card-inner-border-radius);
    }

    &:last-child {
      border-bottom-width: 0;

      @include border-bottom-radius($card-inner-border-radius);
    }
  }

  > .card-header + .list-group,
  > .list-group + .card-footer {
    border-top: 0;
  }
}

.card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: $card-spacer-x;
  color: $card-color;
}

// Card deck
.card-deck {
  .card {
    margin-bottom: $card-deck-margin;
  }

  @include media-breakpoint-up(sm) {
    display: flex;
    flex-flow: row wrap;
    margin-right: calc(#{$card-deck-margin} * -1);
    margin-left: calc(#{$card-deck-margin} * -1);

    .card {
      flex: 1 0 0%;
      margin-right: $card-deck-margin;
      margin-bottom: 0; // Override the default
      margin-left: $card-deck-margin;
    }
  }
}

// Card groups
.card-group {
  > .card {
    margin-bottom: $card-group-margin;
  }

  @include media-breakpoint-up(sm) {
    display: flex;
    flex-flow: row wrap;

    > .card {
      flex: 1 0 0%;
      margin-bottom: 0;

      + .card {
        margin-left: 0;
        border-left: 0;
      }

      // Handle rounded corners
      @if $enable-rounded {
        &:not(:last-child) {
          @include border-right-radius(0);

          .card-img-top,
          .card-header {
            border-top-right-radius: 0;
          }

          .card-img-bottom,
          .card-footer {
            border-bottom-right-radius: 0;
          }
        }

        &:not(:first-child) {
          @include border-left-radius(0);

          .card-img-top,
          .card-header {
            border-top-left-radius: 0;
          }

          .card-img-bottom,
          .card-footer {
            border-bottom-left-radius: 0;
          }
        }
      }
    }
  }
}

// Columns
.card-columns {
  .card {
    margin-bottom: $card-columns-margin;
  }

  @include media-breakpoint-up(sm) {
    column-count: $card-columns-count;
    column-gap: $card-columns-gap;
    orphans: 1;
    widows: 1;

    .card {
      display: inline-block;
      width: 100%;
    }
  }
}

// Accordion
.accordion {
  overflow-anchor: none;

  > .card {
    overflow: hidden;

    &:not(:last-of-type) {
      border-bottom: 0;

      @include border-bottom-radius(0);
    }

    &:not(:first-of-type) {
      @include border-top-radius(0);
    }

    > .card-header {
      @include border-radius(0);

      margin-bottom: calc(#{$card-border-width} * -1);
    }
  }
}
