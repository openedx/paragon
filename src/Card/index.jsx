export { default } from 'react-bootstrap/Card';
export { default as CardColumns } from 'react-bootstrap/CardColumns';
export { default as CardDeck } from 'react-bootstrap/CardDeck';
export { default as CardImg } from 'react-bootstrap/CardImg';
export { default as CardGroup } from 'react-bootstrap/CardGroup';

export { default as CardGrid } from './CardGrid';
