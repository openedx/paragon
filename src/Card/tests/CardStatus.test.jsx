import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Button from '../../Button';
import Card from '..';
import CardStatus from '../CardStatus';
import WarningFilled from '../../../icons/jsx/WarningFilled';

describe('correct rendering', () => {
  it('renders CardStatus element', () => {
    const tree = renderer.create((
      <CardStatus>Text</CardStatus>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correct base className', () => {
    const wrapper = mount(<CardStatus>Text</CardStatus>);
    expect(wrapper.find('.pgn__card-status').length).toBeGreaterThan(0);
  });
  it('renders body with custom className', () => {
    const className = 'my-class-name';
    const wrapper = mount(<CardStatus className={className}>Text</CardStatus>);
    expect(wrapper.find('.my-class-name').length).toBeGreaterThan(0);
  });
  it('renders with correct variant', () => {
    const variant = 'primary';
    const wrapper = mount(<CardStatus variant={variant}>Text</CardStatus>);
    expect(wrapper.find('.pgn__card-status__primary').length).toBeGreaterThan(0);
  });
  it('renders with icon', () => {
    const icon = WarningFilled;
    const wrapper = mount(<CardStatus icon={icon}>Text</CardStatus>);
    expect(wrapper.find('.pgn__icon').length).toBeGreaterThan(0);
  });
  it('renders with title', () => {
    const titleText = 'Hello world!';
    render(<CardStatus title={titleText}>Text</CardStatus>);
    screen.getByText(titleText);
  });
  it('renders skeleton when card is in loading loading state', () => {
    render((
      <Card isLoading>
        <CardStatus>Text</CardStatus>
      </Card>
    ));
    screen.getByTestId('card-status-skeleton');
  });
  it('renders actions', () => {
    render((
      <CardStatus
        actions={<Button>Learn more</Button>}
      >
        Text
      </CardStatus>
    ));
    screen.getByText('Learn more');
  });
});
