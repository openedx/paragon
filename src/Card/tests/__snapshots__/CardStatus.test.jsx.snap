// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct rendering renders CardStatus element 1`] = `
<div
  className="pgn__card-status pgn__card-status__warning"
>
  <div
    className="pgn__card-status__content"
  >
    <div
      className="pgn__card-status__message-content"
    >
      Text
    </div>
  </div>
</div>
`;
